key-manager/src/lib.rs

```diff
@@ -0,0 +1,346 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! A crate used to derive keys useful for the Oxide control plane
+
+use std::collections::BTreeMap;
+
+use async_trait::async_trait;
+use hkdf::Hkdf;
+use secrecy::{ExposeSecret, Secret};
+use sha3::Sha3_256;
+use sled_hardware::DiskIdentity;
+use zeroize::{Zeroize, ZeroizeOnDrop};
+
+/// Input Key Material
+///
+/// This should never be used directly, and always wrapped in a `Secret<Ikm>`
+/// upon construction. We sparate the two types, because a `Secret` must contain
```

Comment:
If this is true, why expose this type as `pub`? Why not just define a pub type alias to the secret, and keep this private?

Reply:
Great call. This should not be `pub` at all. Neither should `Aes256GcmDiskEncryptionKey`. I made both of them private and added accessors to their versioned counterparts which are the types a user sees.

Reply:
Fixed in [79cf577](https://github.com/oxidecomputer/omicron/pull/2990/commits/79cf577eda5264af8622c8e48299c706b2a89a8e)